src/dataflow/src/source/s3.rs

```diff
@@ -580,16 +592,69 @@ async fn download_object(
         }
     };
 
+    // If the Content-Type does not match the compression specified for this
+    // source, emit a debug warning messages and ignore this object
```

Comment:
`debug` level messages don't show up in logs by default. Since this will definitely lead to missing data/incorrect results it should be an `error`-level log.

Reply:
This was the genesis for my question surrounding whether or not we should add a `STRICT` mode. If we're going to make these error messages, it seems better to add them to the error-stream rather than logging them at error level. I'll go ahead and make these errors -- we can relax them later if people are running into issues.

Reply:
Oh, except that anyone using Confluent's S3 Connector Type is going to run into errors if we make this change. I don't think an error message is the right call here.

Reply:
Right, based on the conversation in the design doc this should probably continue being a debug level log. I'm not sure if when I first looked at this a mismatched content-type meant that we didn't process the object. Certainly with the current code -- where we _do_ process the object no matter what -- the debug level logs make sense.